components/reddit/sources/new-link-on-a-subreddit/new-link-on-a-subreddit.js

```diff
@@ -0,0 +1,84 @@
+const reddit = require("../../reddit.app.js");
+const get = require("lodash.get");
+
+module.exports = {
+  key: "new-link-on-a-subreddit",
+  name: "New Link on a subreddit",
+  description: "Emmits a new link submitted to a subreddit",
+  version: "0.0.2",
+  props: {
+    reddit,
+    timer: {
+      label: "Polling schedule",
+      description:
+        "Pipedream poll for new links and emit accordingly with this schedule.",
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 10, // by default, run every 10 minute
+      },
+    },
+    subreddit: {
+      type: "string",
+      label: "Subreddit",
+      description: "Enter subreddit name to listen for new links.",
+      default: "redditdev",
+    },
+    db: "$.service.db",
+  },
+  hooks: {
+    async deploy() {
+      // Emmits all existing events for the first time.
+      var after = null;
+      do {
+        const reddit_things = await this.reddit.getNewSubredditLinks(
+          null,
+          this.subreddit
+        );
+        if (get(reddit_things, "data.after") !== undefined) {
+          after = reddit_things.data.after;
```

Comment:
If you implement pagination in the `getNewSubredditLinks` method, just make sure that you modify this to the max timestamp for all items returned across all pages.